web/thesauruses/c/17/strings.json

```diff
@@ -0,0 +1,353 @@
+{
+  "meta": {
+    "language": "c",
+    "language_version": "17",
+    "language_name": "C",
+    "structure": "strings"
+  },
+  "concepts": {
+    "is_primitive_or_not": {
+      "code": "no",
+      "name": "Is this a built-in type in this language?"
+    },
+    "import": {
+      "code": "#include<string.h>",
+      "name": "Import the string class"
+    },
+    "default_string_byte_encoding": {
+      "code": "ASCII UTF-8",
+      "name": "Default byte encoding (ex: ASCII UTF-8, UTF-16, etc.)"
+    },
+    "create_new_string": {
+      "code": [
+        "char string[] = \"hello\";",
+        "char string[] = {'h','e','l','l','o','\\0'};"
+      ],
+      "name": "Create new string"
+    },
+    "create_multiline_string": {
+      "not-implemented": "true",
+      "name": "Create new multi-line string"
+    },
+    "assign_new_string": {
+      "code": [
+        "char s[] = \"hello\";",
+        "char st[] = s;",
+        "strcpy(st,s);",
+        "printf(\"%s\",st);"
+      ],
+      "name": "Assign string from another string"
+    },
+    "destroy_string": {
+      "not-implemented": "true",
+      "name": "Destroy string"
+    },
+    "length_of_string": {
+      "code": [
+        "chat str[] = \"hello\";",
+        "printf(\"%s\",strlen(s));"
+      ],
+      "name": "Length of string"
+    },
+    "max_length_of_string": {
+      "not-implemented": "true",
+      "name": "Maximum length of string"
+    },
+    "clear_string": {
+      "code": [
+        "char s[] = \"hello\";",
+        "strcpy(s,\"\");"
+      ],
+      "name": "Clear/empty string"
+    },
+    "is_empty": {
+      "code": [
+        "int result = strcmp(s,\"\");",
+        "printf(\"%d\",result);"
+      ],
+      "name": "Is string empty?"
+    },
+    "concatenate_two_strings": {
+      "code": [
+        "char str1[] = \"hello\";",
+        "char str2[] = \" world\"",
+        "strcat(str1,str2);",
+        "puts(str1);"
+      ],
+      "name": "Concatenate/join two strings"
+    },
+    "concatenate_many_strings": {
+      "not-implemented": "true",
+      "name": "Concatenate/join many strings"
+    },
+    "is_all_alphabetical": {
+      "not-implemented":"true",
+      "comment": [
+        "//indirect implementation",
+        "//#include<ctype.h>",
+        "char str[100];",
+        "gets(str);",
+        "for(int i=0;i<strlen(str);i++)",
+        " if(isalpha(str[i]))", 
+           "count+=1;",
+        "if(count==strlen(str)) printf(\"all alphabetical characters\");"
+      ],
```

Comment:
If it's not implemented, you shouldn't include an algorithm to do it since this is a language reference tool.